project-management/src/main/java/life/qbic/projectmanagement/application/measurement/MeasurementService.java

```diff
@@ -317,54 +238,31 @@ protected List<MeasurementId> performRegistration(
   private List<MeasurementId> performRegistrationNGS(
       List<MeasurementMetadata> measurementMetadataList,
       ProjectId projectId) {
-    List<NGSMeasurementMetadata> ngsMeasurements = new ArrayList<>();
+    List<NGSMeasurementMetadata> ngsMeasurementMetadata = new ArrayList<>();
     for (MeasurementMetadata measurementMetadata : measurementMetadataList) {
       if (measurementMetadata instanceof NGSMeasurementMetadata) {
-        ngsMeasurements.add((NGSMeasurementMetadata) measurementMetadata);
+        ngsMeasurementMetadata.add((NGSMeasurementMetadata) measurementMetadata);
```

Comment:
As of Java 16 you do not need to cast here. You can directly declare your variable of type in the if statement.
```java
if (measurementMetadata instanceof NGSMeasurementMetadata ngsMetadata) {
  ngsMeasurementMetadata.add(ngsMetadata);
```
https://openjdk.org/jeps/394

Reply:
The same can be done for the proteomics case ;)

Reply:
Good point!